### PR TITLE
Handle officer and dept mismatch when tagging

### LINKED ISSUE #325: Application should prevent users from associating faces with wrong department
1. User is identifying images associated with Berkeley Police Department
2. User mistypes OpenOversight ID
3. ID is of a different department
4. App should prevent user from entering the wrong ID by flashing a message ``The officer is not in {{ department.name }}. Are you sure that's the correct OpenOversight ID?``

This also should have a unit test

### PR DESCRIPTION
**Pull request by leifan89**
## Status

Ready for review

## Description of Changes

Fixes #325  .

Changes proposed in this pull request:

 - If the tagged officer is with the wrong department, do not proceed
 - Added unit test
 - 

## Tests and linting

 - [x] I have rebased my changes on current `develop`

 - [x] pytests pass in the development environment on my local machine

 - [x] `flake8` checks pass

### CODE REVIEW

**Comment by leifan89** [comment]
(Note that flake8 has flagged lines that were not touched in this PR)

File: OpenOversight/app/main/views.py [hunk 1]
```diff
@@ -636,10 +636,10 @@ def leaderboard():
 def label_data(department_id=None, image_id=None):
     jsloads = ['js/cropper.js', 'js/tagger.js']
     if department_id:
-        department = Department.query.filter_by(id=department_id).one()
+        department = Department.query.filter_by(id=department_id).first()
```
Comment (abandoned-prototype) [inline OpenOversight/app/main/views.py:639]: Just curious @leifan89, can you elaborate why this change is necessary? Are there situations that an invalid department_id is provided that shouldn't result in an error?

Comment (leifan89) [reply OpenOversight/app/main/views.py:639]: Thanks for the review!

Unfortunately I'm having issues replicating the problem that I encountered before with `one()` instead of `first()`... do you think that it would be better to throw an error explicitly instead of just swallowing it? Happy to restore back to the original `first()` 🙂

File: OpenOversight/app/main/views.py [hunk 2]
```diff
@@ -667,6 +667,8 @@ def label_data(department_id=None, image_id=None):
                          .filter(Face.original_image_id == form.image_id.data).first()
         if not officer_exists:
             flash('Invalid officer ID. Please select a valid OpenOversight ID!')
+        elif department and officer_exists.department_id != department_id:
+            flash(f'The officer is not in {department.name}. Are you sure that is the correct OpenOversight ID?')
```
Comment (abandoned-prototype) [inline OpenOversight/app/main/views.py:671]: Flake8 fails here because python 3.5 does not support the "f-string" string formatting. I am not sure if we still use python 3.5 in production or anywhere, so I am definitely in favor of dropping 3.5 support, but until we do you might need to use `.format`

Comment (leifan89) [reply OpenOversight/app/main/views.py]: Thank you for the pointer! I've updated that :)

File: OpenOversight/app/main/views.py (hunk 1 above)
Comment (dismantl) [reply OpenOversight/app/main/views.py]: I think leaving it as `one()` makes more sense, since there shouldn't be any cases in which there are multiple departments with the same department_id that would be returned by the query.

Comment (leifan89) [reply OpenOversight/app/main/views.py]: So I did manage to replicate the original issue that caused me to change it to `first()`...

As it turns out, it was the test that I had added, in which I tested the scenario of an officer with a mismatched department. In that case, since the query with both that department and that officer fails to find anything (the mismatch case), `one()` fails since it expects exactly 1 result, not 0.

Comment (abandoned-prototype) [reply OpenOversight/app/main/views.py]: @dismantl this particular case is fixed and no longer in the code

**Comment by abandoned-prototype** [comment]
The instance of `first` on views.py:642 makes sense to me since it is possible to provide image_id and department_id that do not match, in which case I don't think we want to trigger a 500 error

**Comment by dismantl** [comment]
> The instance of `first` on views.py:642 makes sense to me since it is possible to provide image_id and department_id that do not match, in which case I don't think we want to trigger a 500 error

Sorry you're right, I was looking at the outdated code on line 639.

**Comment by dismantl** [comment]
Just merged in develop and tested, working great.
